flocker/node/agents/blockdevice.py

```diff
@@ -120,10 +247,48 @@ class BlockDeviceVolume(PRecord):
     :ivar int size: The size, in bytes, of the block device.
     :ivar unicode host: The IP address of the host to which the block device is
         attached or ``None`` if it is currently unattached.
+    :ivar UUID dataset_id: The Flocker dataset ID associated with this volume.
     """
     blockdevice_id = field(type=unicode, mandatory=True)
     size = field(type=int, mandatory=True)
     host = field(type=(unicode, type(None)), initial=None)
+    dataset_id = field(type=UUID, mandatory=True)
+
```

Comment:
If these are specific to loopback volume then make them utility functions, not class methods for a class that will be used by all backends.